### PR TITLE
[QuickStart] Yosys/Parmys Update

### LINKED ISSUE #2739: Update VTR quick start guide to use yosys / parmys
The VTR quick start guide uses Odin II as the synthesis tool, but the yosys/parmys flow is now the default VTR flow. We should update the quick start to use yosys/parmys.  The Odin II section should be moved down to a later, optional part of the quick start that just describe how to run Odin II (as the primary example of yosys/parmys would have already covered the whole flow).

#### Proposed Behaviour
Change all mention of Odin II to the equivalent parmys / yosys flow commands.  Move the Odin II command(s) down to the end of the quick start under an Odin II section.

#### Current Behaviour
Odin II is the only synthesis tool described in the quick start.

### PR DESCRIPTION
**Pull request by ZohairZaidi**
#### Description

This pull request updates the VTR Quick Start Guide by replacing Odin II as the primary synthesis tool with the Yosys/Parmys flow. I have also updated the graphic images
#### Related Issue

closes #2739  

#### Checklist:

- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

### CODE REVIEW

File: doc/src/quickstart/index.rst
Comment (vaughnbetz) [inline doc/src/quickstart/index.rst:271]: parmys works a little differently than Odin-II; it shoul have already optimized and technology mapped the design.  I suggest you take a look at the .blif file that comes out of parmys and see if it is mostly LUTs (.names), with some having up to 6 inputs.  If so, it has been tech-mapped. We can still show people how to run abc, but should modify the text to say parmys already called abc internally to optimize the design, but if we'd like to try some further optimization or technology mapping options we can also directly invoke abc on the parmys output.

File: doc/src/quickstart/index.rst
Comment (vaughnbetz) [inline doc/src/quickstart/index.rst:294]: Take a look at the two files and see if this is true.  Are the number of .names significantly different?  Confirm they are different a bit at least.  We shouldn't say it is significantly simpler / optimized unless it is.

File: doc/src/quickstart/index.rst
Comment (vaughnbetz) [inline doc/src/quickstart/index.rst:463]: Add (at the very start of this section):

VTR includes a second synthesis tool, ODIN II.  Below we explain how to run this alternative synthesis. flow.

File: doc/src/quickstart/index.rst
Comment (vaughnbetz) [inline doc/src/quickstart/index.rst:308]: I think this is only needed because we directly ran ABC on the parmys output, and that should be stated (when run directly, ABC does not preserve clocks so we re-insert them in the re-optimized design with the restore_multiclock_latch.pl script).

File: doc/src/quickstart/index.rst
Comment (vaughnbetz) [inline doc/src/quickstart/index.rst:313]: Does run_vtr_flow directly call abc and the restore_multiclock_latch.pl script when run with parmys?  If not, I think we should leave ABC and this script out of the parmys section, and just refer to them in the Odin II section (which does need them).

**Review by vaughnbetz — CHANGES_REQUESTED** [review]
Thanks @ZohairZaidi !  
I have some comments about the parmys flow (I think we've listed some unnecessary steps for it) that you should take a look at closely. Adding @amirarjmand93 and @KennethKent to review; in case @AlexandreSinger has time to review I've tagged him too

File: doc/src/quickstart/index.rst
Comment (vaughnbetz) [reply doc/src/quickstart/index.rst:294]: Thanks.  I think we should just simplify then to show running parmys in this section, without abc or the clock relabelling (we can leave those in the Odin-II section). If you can confirm the clock relabeling is *not* called for the parmys flow in run_vtr_flow that would be ideal -- then we can confidently remove it from this section.

Comment (ZohairZaidi) [reply doc/src/quickstart/index.rst:294]: sure thing. It does not appear to be called in run_vtr_flow. 

My initial confusion stemmed from reading the Parmys paper, which states, "the Parmys pass is called to perform partial mapping according to the target architecture, and then a series of optimization and logic mappings are performed to convert all the remaining Yosys internal types to primitives. Next, the output is passed to ABC and VPR stages."

This led me to believe that running only the Parmys part of the flow would require a separate invocation of ABC since we are executing the command with both -start parmys and -end parmys, it explicitly skips the ABC.

I will update the PR and also wait for final confirmation by the added members

**Comment by ZohairZaidi** [comment]
Thanks @vaughnbetz!, I have addressed your suggestions. Will wait for the added members to review as well.

File: doc/src/quickstart/index.rst
Comment (AlexandreSinger) [inline doc/src/quickstart/index.rst:235]: Is this line still correct? Is ABC being used when manually running the VTR flow?

File: doc/src/quickstart/index.rst
Comment (AlexandreSinger) [inline doc/src/quickstart/index.rst:349]: Same comment as above, is ABC log files still generated?

File: doc/src/quickstart/index.rst
Comment (AlexandreSinger) [inline doc/src/quickstart/index.rst:371]: Are these abc blif files still there?

File: doc/src/quickstart/index.rst
Comment (AlexandreSinger) [inline doc/src/quickstart/index.rst:279]: Why are you passing the full file path? If someone is following your steps correctly, should they not already be in the `~/vtr_work/quickstart/blink_manual/` folder? If you do want to make it explicit, I would recommend creating a `$WRK_DIR` variable, similar to `$VTR_ROOT` instead of passing the whole path.

File: doc/src/quickstart/index.rst
Comment (AlexandreSinger) [inline doc/src/quickstart/index.rst:296]: Same path comment as above.

File: doc/src/quickstart/index.rst
Comment (AlexandreSinger) [inline doc/src/quickstart/index.rst:396]: I would say "we explain how to manually run". Just to be clear that this is not the recommended way, but its possible if you want to avoid the scripts.

File: doc/src/quickstart/index.rst
Comment (AlexandreSinger) [inline doc/src/quickstart/index.rst:409]: This is confusing since the line above tells the user to change directory to `~/vtr_work/quickstart/blink_manual` and now you are saying to use the make command. This should be in the VTR directory. I would move the build instructions above the prior code block and provide the `cd` command to go to the correct directory.

**Review by AlexandreSinger — COMMENTED** [review]
@ZohairZaidi Thank you for these changes! I like how you are running Yosys. I think using the vtr_flow script just on the Yosys step is perfectly fine since running Yosys directly is such a pain. The main goal is to know how to make the .blif file.

I have left some comments below, but overall the documentation is very good.

File: doc/src/quickstart/index.rst
Comment (ZohairZaidi) [reply doc/src/quickstart/index.rst:235]: it is used when we used synthesize with ODIN.

Comment (AlexandreSinger) [reply doc/src/quickstart/index.rst:235]: Yes, but these bullet points are in the "Manually Running the VTR Flow" section; not the with ODIN section.

Comment (vaughnbetz) [reply doc/src/quickstart/index.rst:235]: Agreed. This should be refactored to make clear you either need to run Odin II + ABC, or parmys (which will internally call ABC itself).

**Comment by vaughnbetz** [comment]
Adding @soheilshahrouz in case he can help figure out the required flow ... we need to document this well in the quickstart as Mohamed and I were confused about it.

**Comment by ZohairZaidi** [comment]
I agree @vaughnbetz, I plan to work on it this reading week. Please let me know what you were confused about.

File: doc/src/quickstart/index.rst
Comment (ZohairZaidi) [inline doc/src/quickstart/index.rst:431]: I was wondering if this is necessary. I would assume that if they opt with ODIN they would skip the synthesizing with Parmys section so I left it here.

File: doc/src/quickstart/index.rst
Comment (vaughnb-cerebras) [reply doc/src/quickstart/index.rst]: I think you were right and I was wrong @ZohairZaidi !  After talking to Ken Kent at UNB, it does seem we need to run ABC and the clock add/removal script after Parmys.  So I think your original edits were right.  Can you run the flow to check exactly what happens in run_vtr_flow with parmys?   I think it runs ABC and the clock recovery script after all, so we should document them again.

I also think we should change the order of the text.  After saying how to run vpr, I think we should say how to use run_vtr_flow with parmys to synthesize, place and route a circuit. Then we can have a "running manually" section to dig more into how to run each tool.  Finally we can have the existing Odin II section.

**Comment by ZohairZaidi** [comment]
Hi @vaughnbetz, let me know how this is. I changed the order and made it smoother for the user to follow.

File: doc/src/quickstart/index.rst
Comment (vaughnbetz) [inline doc/src/quickstart/index.rst:165]: I think we should rename this to "Running VPR Manually"
and then add a first sentence after the heading that says
"Sometimes you may wish to run only the vpr (placement, routing and timing analysis) parts of the flow rather than the full VTR flow (which includes synthesis). To show how to do this, let's now try taking ..."

**Review by vaughnbetz — APPROVED** [review]
This looks good, thanks!  I just have a couple of minor comments / changes to link up the first couple of sections.

**Comment by vaughnbetz** [comment]
Looks good, thanks!

**Pull request merged**
